app/middleware/simplified-account/simplified-account-strategy.middleware.test.js

```diff
@@ -0,0 +1,209 @@
+      return {
+        getAccountByServiceIdAndAccountType: connectorGetAccountMock,
+        getStripeAccountSetup: paymentProvider === 'stripe'
+          ? sinon.stub().resolves(stripeAccountSetupFixture.buildGetStripeAccountSetupResponse())
+          : undefined
+      }
+    }
+  }
+
+  const simplifiedAccountStrategy = proxyquire(
+    path.join(__dirname, './simplified-account-strategy.middleware'),
+    { '../../services/clients/connector.client.js': connectorMock }
+  )
+
+  return {
+    simplifiedAccountStrategy,
+    connectorGetAccountMock
+  }
+}
+
+describe('Middleware: getSimplifiedAccount', () => {
+  it('should set gateway account and service on request object', async () => {
+    const { simplifiedAccountStrategy, connectorGetAccountMock } = setupSimplifiedAccountStrategyTest({
+      gatewayAccountID: '1',
+      gatewayAccountExternalId: A_GATEWAY_EXTERNAL_ID,
+      paymentProvider: 'worldpay',
+      serviceExternalId: A_SERVICE_EXTERNAL_ID,
+      accountType: 'test'
+    })
+    await simplifiedAccountStrategy(req, res, next)
+
+    sinon.assert.calledOnce(next)
+    expect(connectorGetAccountMock.called).to.equal(true)
+    expect(connectorGetAccountMock.calledWith({
+      serviceId: A_SERVICE_EXTERNAL_ID,
+      accountType: 'test'
+    })).to.equal(true)
+
+    expect(req.account.external_id).to.equal(A_GATEWAY_EXTERNAL_ID)
+    expect(req.service.externalId).to.equal(A_SERVICE_EXTERNAL_ID)
+  })
+  it('should error if service external ID or gateway account type cannot be resolved from request parameters', async () => {
+    const { simplifiedAccountStrategy } = setupSimplifiedAccountStrategyTest({
+      gatewayAccountID: '1',
+      gatewayAccountExternalId: A_GATEWAY_EXTERNAL_ID,
+      paymentProvider: 'worldpay',
+      serviceExternalId: A_SERVICE_EXTERNAL_ID,
+      accountType: 'test'
+    })
+    req.params['serviceExternalId'] = undefined
+    req.params['accountType'] = undefined
+
+    await simplifiedAccountStrategy(req, res, next)
+
+    const expectedError = sinon.match.instanceOf(Error)
+      .and(sinon.match.has('message', 'Could not resolve service external ID or gateway account type from request params'))
+    sinon.assert.calledWith(next, expectedError)
+  })
+  it('should error if gateway account lookup fails for account type', async () => {
+    const { simplifiedAccountStrategy } = setupSimplifiedAccountStrategyTest({
+      gatewayAccountID: '1',
+      serviceExternalId: A_SERVICE_EXTERNAL_ID,
+      accountType: 'test',
+      errorCode: '404'
+    })
+    await simplifiedAccountStrategy(req, res, next)
+
+    const expectedError = sinon.match.instanceOf(Error)
+      .and(sinon.match.has('message', 'getGatewayAccountByServiceIdAndAccountType failed for provided parameters'))
+    sinon.assert.calledWith(next, expectedError)
+  })
+  describe('extend gateway account data with disableToggle3ds field', () => {
+    ['worldpay', 'smartpay', 'epdq'].forEach(function (paymentProvider) {
```

Comment:
Do we need smartpay and epdq here?

Reply:
we can always add them back if we need to later, removed